### ISSUE TITLE
Vue file type configuration setting does not be saved

### ISSUE THREAD

**Issue by saidbakr** [Microsoft/vscode#59809]
Issue Type: <b>Bug</b>

1. Open any .vue component file.
2. From the bottom status blue bar, click on the "Vue" text with tooltip "Select Language Mode.
3. From the menu select "Configure Vue language based settings..."
4. In the search settings field, enter "match"
5. In the item titled "Editor: Match Brackets". Check the checkbox "Highlight matching brackets..."
6. Return back to any .vue file, the matched brackets feature is working fine.
7. Close or exit the editor.
8. Open the editor again.
9. Open any .vue file again, you will notice that th editor loses the feature, i.e brackets in this file type don't be matched.
10. repeat from (2) to (5), you will find that the option that checked in the previous session, becomes unchecked.

VS Code version: Code 1.27.2 (f46c4c469d6e6d8c46f268d1553c5dc4b475840f, 2018-09-12T07:11:15.873Z)
OS version: Linux x64 4.4.0-135-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-2450M CPU @ 2.50GHz (4 x 2932)|
|GPU Status|2d_canvas: unavailable_software<br>checker_imaging: disabled_off<br>flash_3d: unavailable_software<br>flash_stage3d: unavailable_software<br>flash_stage3d_baseline: unavailable_software<br>gpu_compositing: unavailable_software<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: unavailable_software<br>video_decode: unavailable_software<br>video_encode: unavailable_software<br>webgl: enabled_readback<br>webgl2: unavailable_off|
|Load (avg)|0, 0, 1|
|Memory (System)|7.76GB (0.16GB free)|
|Process Argv|/usr/share/code/code --unity-launch|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (30)</summary>

Extension|Author (truncated)|Version
---|---|---
EditorConfig|Edi|0.12.4
crane|Hvy|0.3.8
indenticator|Sir|0.6.0
php-formatter|Sop|0.2.4
html-snippets|abu|0.2.1
vscode-intelephense-client|bme|0.8.8
laravel-goto-view|cod|1.2.1
compulim-vscode-closetag|Com|1.2.0
vscode-html-css|ecm|0.2.0
vscode-great-icons|emm|2.1.42
php-debug|fel|1.12.6
php-intellisense|fel|2.3.5
auto-close-tag|for|0.5.6
beautify|Hoo|1.4.4
easy-icons|jam|0.3.1
php-namespace-resolver|Meh|1.1.0
vsliveshare|ms-|0.3.735
color-highlight|nau|2.3.0
snippet-creator|nik|0.0.5
vetur|oct|0.12.7
laravel-blade|one|1.18.0
laravel-extension-pack|one|0.4.0
laravel5-snippets|one|1.5.0
advanced-new-file|pat|1.2.0
laravel-helpers|raf|0.2.2
subtle-brackets|raf|3.0.0
laravel-artisan|rya|0.0.22
laravel-5-snippets|sac|2.0.1
guides|spy|0.9.2
highlight-matching-tag|vin|0.4.4

(1 theme extensions excluded)

</details>

**Comment by saidbakr** [comment]
> (Experimental duplicate detection)
> Thanks for submitting this issue. Please also check if it is already covered by an existing one, like:
> 
> * [Regions don't appear to work in .vue files (#39939)](https://www.github.com/Microsoft/vscode/issues/39939)
> * [word wrap in *.vue's <style> long rule (#59698)](https://www.github.com/Microsoft/vscode/issues/59698)

No it is not like the regarded issues above.

**Comment by RMacfarlane** [comment]
Can you open the `settings.json` file and report its contents after step 5 and 8? 

![screen shot 2018-10-03 at 11 55 44 am](https://user-images.githubusercontent.com/3672607/46432536-5b932780-c703-11e8-82d4-95a0f9161ca4.png)

I can repro that language-specific settings aren't being written, but not sure why behavior changes after restarting

**Comment by saidbakr** [comment]
After step 5 the contents of `settings.json` are:
![image](https://user-images.githubusercontent.com/467554/46435679-34664580-c757-11e8-8897-7edf3c5c1b3b.png)

After step 8, i.e after restarting the program:
![image](https://user-images.githubusercontent.com/467554/46435860-b22a5100-c757-11e8-8125-e5fbce46074a.png)

### Very Important Notice

This issue is clearly found in a pure Vue.js project. However, when I close the program and then using right on a folder that includes Laravel project which contains some .vue components files. The matching brackets works fine with this project and the following is `settings.json` screen shot with it:
![image](https://user-images.githubusercontent.com/467554/46436184-95dae400-c758-11e8-89aa-439a941384a6.png)

**Comment by roblourens** [comment]
In the last one, it looks like there's also a match in workspace settings, so maybe that project has the setting overridden in workspace settings.

Could there be some extension toggling this setting? Would editorConfig do that? Can you repeat this starting with the `--disable-extensions` flag to test?

**Comment by saidbakr** [comment]
@roblourens In the Vue project, I search for extensions with "match" keyword and I found an installed extension called [`rafamel.subtle-brackets`](https://marketplace.visualstudio.com/items?itemName=rafamel.subtle-brackets). When I disabled it, everything worked fine even after restarting the editor!

The question is why it toggled the setting value in Vue project while it does not toggled it Laravel project?!

**Comment by roblourens** [comment]
Interesting, thanks for finding that extension.

Probably because of

> In the last one, it looks like there's also a match in workspace settings, so maybe that project has the setting overridden in workspace settings.
